deploy/awstools/awstools.py

```diff
@@ -36,6 +36,60 @@ def depaginated_boto_query(client, operation, operation_params, return_key):
         return_values_all += page[return_key]
     return return_values_all
 
+def get_manager_instance_id():
+    """Get current manager instance id, if applicable.
+
+    Returns:
+        A ``str`` of the instance id or ``None``
+    """
+    # This takes multiple minutes without a timeout from the CI container. In
+    # practise it should resolve nearly instantly on an initialized EC2 instance.
+    curl_connection_timeout = 10
+    with settings(warn_only=True), hide('everything'):
```

Comment:
This code must be duplicated in multiple places in our python codebase -- can it not be DRYed out?

Reply:
I grepped for `169.254...` and only found the two matches in this file? Were you referring to something else?